apps/crn-server/src/routes/research-outputs.route.ts

```diff
@@ -45,44 +47,61 @@ export const researchOutputRouteFactory = (
   );
 
   researchOutputRoutes.post('/research-outputs', async (req, res) => {
-    const { body, loggedInUser } = req;
+    const { body, loggedInUser, query } = req;
     const createRequest = validateResearchOutputPostRequestParameters(body);
     validateResearchOutputPostRequestParametersIdentifiers(createRequest);
 
-    if (
-      !loggedInUser ||
-      !hasCreateUpdateResearchOutputPermissions(
-        loggedInUser,
-        createRequest.teams,
-      )
-    ) {
+    const permissions = getUserPermissions(
+      loggedInUser as UserResponse,
+      'teams',
+      createRequest.teams,
+    );
+
+    const options = validateResearchOutputRequestQueryParameters(query);
+    const published = options.published ?? true;
+
+    const isRequestAllowed =
+      (permissions.saveDraft && !published) ||
+      (permissions.publish && published);
+
+    if (!loggedInUser || !isRequestAllowed) {
       throw Boom.forbidden();
     }
 
-    const researchOutput = await researchOutputController.create({
-      ...createRequest,
-      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-      createdBy: loggedInUser!.id,
-    });
+    const researchOutput = await researchOutputController.create(
+      {
+        ...createRequest,
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        createdBy: loggedInUser!.id,
+      },
+      published,
+    );
 
     res.status(201).json(researchOutput);
   });
 
   researchOutputRoutes.put(
     '/research-outputs/:researchOutputId',
     async (req, res) => {
-      const { body, params, loggedInUser } = req;
+      const { body, params, loggedInUser, query } = req;
       const { researchOutputId } = validateResearchOutputParameters(params);
       const updateRequest = validateResearchOutputPutRequestParameters(body);
       validateResearchOutputPostRequestParametersIdentifiers(body);
 
-      if (
-        !loggedInUser ||
-        !hasCreateUpdateResearchOutputPermissions(
-          loggedInUser,
-          updateRequest.teams,
-        )
-      ) {
+      const permissions = getUserPermissions(
```

Comment:
same as above, rather than split into stages i would just pass on the `published` option (should be `publish` but thats a different comment) to the `hasCreateUpdateResearchOutputPermissions` function which should determine permissions based on the input